pynq/lib/logictools/waveform.py

```diff
@@ -176,24 +173,188 @@ def draw_wavedrom(data):
         ]}
     >>> draw_wavedrom(a)
 
+    More information can be found at:
+    https://github.com/witchard/nbwavedrom
+
+    Parameters
+    ----------
+    data : dict
+        A dictionary of data as shown in the example.
+
+    """
+    data = _dump_json_data(data)
+    phantomjs = _find_phantomjs()
+    if phantomjs:
+        wavedrom_cli = _find_wavedrom_cli()
+        return _draw_phantomjs(data, phantomjs, wavedrom_cli)
+    else:
+        return _draw_javascript(data)
+
+
+def _dump_json_data(data):
+    """Convert the data into Json dump.
+
+    Parameters
+    ----------
+    data : dict
+        A dictionary of the Json formatted data.
+
+    Returns
+    -------
+    str
+        A Json dump of the original data.
+
+    """
+    return json.dumps(data)
+
+
+def _draw_javascript(data):
+    """Display the waveform using the Wavedrom Javascript.
+
+    This method requires 2 javascript files to be present. We get the relative
+    paths for the 2 files in order to proceed.
+    Users can call this method directly to draw any wavedrom data.
+
+    Parameters
+    ----------
+    data : str
+        A dump of a Json formatted data.
+
     """
-    wavedrom_js = '/js/WaveDrom.js'
-    wavedromskin_js = '/js/WaveDromSkin.js'
-    if not os.path.isfile(PYNQ_JUPYTER_NOTEBOOKS + wavedrom_js):
-        raise RuntimeError('Cannot locate {}.'.format(wavedrom_js))
-    if not os.path.isfile(PYNQ_JUPYTER_NOTEBOOKS + wavedromskin_js):
-        raise RuntimeError('Cannot locate {}.'.format(wavedromskin_js))
+    wavedrom_js = 'wavedrom.js'
+    wavedromskin_js = 'wavedromskin.js'
+
+    if not (_is_javascript_present(wavedrom_js) and
+            _is_javascript_present(wavedromskin_js)):
+        _copy_javascripts()
```

Comment:
Why are we copying the Javascript? Why don't we just do what nbwavedrom does and insert the wavedrom code into the notebook output?